### ISSUE TITLE
Telepresence does not use the correct context in every call to kubectl

### ISSUE THREAD

**Issue by exarkun** [telepresenceio/telepresence#365]
### What were you trying to do?

Trying to reproduce the "Quick example" from https://www.telepresence.io/tutorials/docker

```
kubectl run qotm --image=datawire/qotm:1.3 --port=5000 --expose
telepresence --docker-run -i -t alpine /bin/sh
```

### What did you expect to happen?

A shell running in a Docker container running the alpine image where I could poke at a qotm server.

### What happened instead?

Several exceptions reported by telepresence but no shell in a Docker container.  This shows up in the logs:

```
Unable to connect to the server: dial tcp 192.168.99.100:8443: i/o timeout
```

but it's also the first line of output from the telepresence command.

### Automatically included information

Command line: `['/usr/bin/telepresence', '--context', 'staging', '--docker-run', '-i', '-t', 'alpine', '/bin/sh']`
Version: `0.68`
Python version: `3.5.2+ (default, Sep 22 2016, 12:18:14) 
[GCC 6.2.0 20160927]`
kubectl version: `Client Version: v1.6.4`
oc version: `(error: [Errno 2] No such file or directory: 'oc')`
OS: `Linux baryon 4.8.0-59-generic #64-Ubuntu SMP Thu Jun 29 19:38:34 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`
Traceback:

```
Traceback (most recent call last):
  File "/usr/bin/telepresence", line 1785, in get_proxy_cidrs
    get_output([runner.kubectl_cmd, "get", "nodes", "-o", "json"])
  File "/usr/bin/telepresence", line 395, in get_output
    result = unicode(check_output(*args, **kwargs).strip(), "utf-8")
  File "/usr/lib/python3.5/subprocess.py", line 626, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.5/subprocess.py", line 708, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['kubectl', 'get', 'nodes', '-o', 'json']' returned non-zero exit status 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/telepresence", line 257, in call_f
    return f(*args, **kwargs)
  File "/usr/bin/telepresence", line 2379, in go
    ssh,
  File "/usr/bin/telepresence", line 1978, in run_docker_command
    runner, args, remote_info, remote_env["KUBERNETES_SERVICE_HOST"]
  File "/usr/bin/telepresence", line 1792, in get_proxy_cidrs
    get_output([runner.kubectl_cmd, "get", "pods", "-o", "json"])
  File "/usr/bin/telepresence", line 395, in get_output
    result = unicode(check_output(*args, **kwargs).strip(), "utf-8")
  File "/usr/lib/python3.5/subprocess.py", line 626, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.5/subprocess.py", line 708, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['kubectl', 'get', 'pods', '-o', 'json']' returned non-zero exit status 1

```

Logs:

```
:44+0000 [twisted.scripts._twistd_unix.UnixAppLogger#info] reactor class: twisted.internet.epollreactor.EPollReactor.
   8.9 TL | [19] ran.
   8.9 TL | [20] Capturing: (['kubectl', '--context', 'staging', '--namespace', 'default', 'exec', '--container=telepresence-1513177985-1553378-9170', 'telepresence-1513177985-1553378-9170-2313679694-rf6xp', '--', 'python3', '-c', '\nimport socket, sys, json\n\nresult = []\nfor host in sys.argv[1:]:\n    result.append(socket.gethostbyname(host))\nsys.stdout.write(json.dumps(result))\nsys.stdout.flush()\n'],)...
  10.3 TL | [20] captured.
  10.3 TL | [21] Capturing: (['kubectl', 'get', 'nodes', '-o', 'json'],)...
Unable to connect to the server: dial tcp 192.168.99.100:8443: i/o timeout
  40.4 TL | Failed to get nodes: Command '['kubectl', 'get', 'nodes', '-o', 'json']' returned non-zero exit status 1
  40.4 TL | [22] Capturing: (['kubectl', 'get', 'pods', '-o', 'json'],)...
Unable to connect to the server: dial tcp 192.168.99.100:8443: i/o timeout

```

**Comment by exarkun** [comment]
Looking at the logs, I see that some of the `kubectl` child processes have `--context staging` (copied from my telepresence command) but others don't.  It looks like those that don't are using the active context (which happens to point to a stopped minikube) and those are the ones with a connection failure.

**Comment by exarkun** [comment]
I worked around this by changing my active context to `staging` so the `--context staging` argiument isn't necessary:

```
kubectl config use-context staging
```

**Comment by ark3** [comment]
We must replace direct subprocess calls to `kubectl` with calls to `Runner.get_kubectl(...)`and friends.

**Comment by exarkun** [comment]
Another area of the code bitten by this seems to be service IP range detection.  When I had one context active and passed another to `telepresence --context ...`, telepresence detected and proxied the service ip range from the active context instead of the one I specified (fortunately they were different so I noticed because nothing worked :smile:)

**Comment by ark3** [comment]
We should think about how to test for this issue when we fix it, or at least how to discourage the accretion of new rogue calls to `kubectl` that are not correct.

**Comment by ark3** [comment]
Some of the non-`runner.py` code imports the `subprocess` module just to grab types, exceptions, and constants (`Popen`, `CalledProcessError`, `STDOUT`). We can modify that code to grab those from `runner.py` instead. Then we can add a lint stage that detects `subprocess` imports outside of that file.

**Comment by ark3** [comment]
- main.py: uses `check_output` before `Runner` exists
- ssh.py: `Popen` as a type
- cli.py: `check_output` as part of crash reporter
- vpn.py: `Popen` for `sudo true`; needs to be fixed, but not relevant to this issue
- container.py: uses `Popen` to launch user's command
- cleanup.py: `Popen` as a type
- local.py: uses `Popen` to launch user's command

But the issues are caused by things like this:
```python
            runner.get_output([
                runner.kubectl_cmd, "get", "pods", "-o", "json"
            ])
```

**Comment by ark3** [comment]
The kubectl stuff should have been fixed in #706. Release coming soon...
